staff/edit-courses/edit-course.js

```diff
@@ -0,0 +1,359 @@
+    }
+}
+
+function toggleCredits(btn) {
+    timeout = setTimeout(function(btn) {
+        if (btn.ariaPressed === 'true') { // Apparently this isn't set immediately. Timeout works
+            document.getElementById("credits_max_separator").style.display = ""; 
+            document.getElementById("max_hours").style.display = ""; 
+        } else {
+            document.getElementById("credits_max_separator").style.display = "none"; 
+            document.getElementById("max_hours").style.display = "none"; 
+        }
+        btn.value = btn.ariaPressed;
+    }, 1, btn);
+}
+
+function dropdownSelect(selection) {
+    timeout = setTimeout(function(selection) {
+        let drop = selection.parentNode.previousElementSibling;
+        let choiceValue = selection.attributes.value.nodeValue;
+        let choiceText = selection.text;
+        drop.value = choiceValue;
+        drop.innerText = choiceText;
+
+        // Hide or show year box if altering a requisite semester
+        if (drop.id == "req-sem") {
+            if (drop.value == 'none') {
+                drop.nextElementSibling.nextElementSibling.style.display='none';
+            } else {
+                drop.nextElementSibling.nextElementSibling.style.display='';
+            }
+        }
+
+    }, 1, selection);
+}
+
+function addReq() {
+    let table = document.getElementById("reqs-table");
+    table.insertRow();
+
+    let current_req = nextReq;
+    fetch("requisite.php?req_num="+current_req).then(
+        response => response.text()
+    ).then(function(text) {
+        table.rows[table.rows.length -1].outerHTML = text;
+        updateReqAutoComplete(current_req);
+    });
+    nextReq++;
+}
+
+function removeReq(btn) {
+    let row = btn.parentNode.parentNode;
+    row.parentNode.removeChild(row);
+}
+
+function populateModal(btn) {
+    let course_code = "";
+    
+    if (btn.innerText.trim() === "Add new course") {
+        course_code = "New";
+    } else {
+        let course_row = btn.parentNode.parentNode;
+        course_code = course_row.children[0].innerText;
+    }
+
+    fetch("edit-course.php?course_code="+course_code).then(
+        response => response.text()
+    ).then(
+        function(text) {
+            document.getElementById("edit-course").innerHTML = text;
+            // Find last req in course to determine how many need updated with autocomplete
+            let req_pos = text.lastIndexOf("reqCode");
+            let req_end = text.indexOf("'", req_pos);
+            let req_id = text.slice(req_pos + "reqCode-".length, req_end);
+            for (let i = 0; i <= req_id; i++) {
+                // Update all reqs with autocomplete functionality
+                updateReqAutoComplete(i);
+            }
+            // Handle submit button manually to prevent enter from pressing default button
+            $('#form-button-submit').click(function(){
```

Comment:
Just for the sake of consistency, we should do this the non-jQuery way like we do elsewhere, i.e.
`document.getElementById("form-button-submit").addEventListener("click", () => ...`